tests/integration_tests/test_hyperopt.py

```diff
@@ -622,3 +633,51 @@ def test_hyperopt_grid_search_more_than_one_sample(csv_filename, tmpdir, ray_clu
 
     with pytest.warns(RuntimeWarning):
         hyperopt(config, dataset=rel_path, output_directory=tmpdir, experiment_name="test_hyperopt")
+
+
+@pytest.mark.distributed
+@pytest.mark.parametrize(
+    "max_concurrent_trials", ["auto", None], ids=["max_concurrent_trials_auto", "no_max_concurrent_trials_specified"]
+)
+def test_hyperopt_with_infer_max_concurrent_trials(csv_filename, tmpdir, max_concurrent_trials, ray_cluster):
```

Comment:
How expensive is it to run this test? My concern is that we already test out that `update_max_concurrent_trials` works as expected in the hyperopt unit tests below. Since this test requires running a full hyperopt experiment, it ends up being an expensive test to run to test out functionality that is already mostly covered elsewhere. This is also in the recent context that we've seen of a lot of timeouts in hyperopt testing.

What do you think?

Reply:
@ShreyaR Great points! 

I agree that we may not need the entire test. My incentive to keep this test is that it ensures that `hyperopt_statistics.json` does actually reflect that this value was set. One very easy way of stripping this test down to the bare bones is to basically remove the check for number of trials run, set time_budget_s to something like 15 seconds, and then just do a check for `max_concurrent_trials` in `hyperopt_statistics.json`. What do you think?